### PR TITLE
Refresh base64 support

### PR DESCRIPTION
**Pull request by shaleh**
I would like to move decode to no allocation as well. However, the
only mechanism upstream uses a Vector which cannot be traded with C.
Once I get information about decode we can merge this.

### CODE REVIEW

File: rust_src/src/base64.rs [hunk 1]
```diff
@@ -157,16 +257,12 @@ fn base64_encode_string(string: LispObject, no_line_break: LispObject) -> LispOb
...
         string.sdata_ptr(),
-        encoded,
         length,
+        encoded,
+        allength,
         no_line_break.is_nil(),
         string.is_multibyte(),
     );
-
-    if encoded_length > allength {
```
Comment (DavidDeSimone) [inline rust_src/src/base64.rs]: Is this check no longer worthwhile?

Comment (shaleh) [reply rust_src/src/base64.rs]: Correct. The base64 code already checked it.

Comment (shaleh) [reply rust_src/src/base64.rs]: I hope to update the upstream crate to export the computation to make our allocation better.

Comment (shaleh) [reply rust_src/src/base64.rs]: Currently the upstream code panics if the size is off. I also hope to get a Result returning API accepted.

Comment (DavidDeSimone) [reply rust_src/src/base64.rs]: Sounds good to me. 👍

**Comment by shaleh** [comment]
While I work on the upstream code I think this round is ok to check in. Any complaints?

**Pull request merged**